pypeit/calibrations.py

```diff
@@ -637,60 +640,69 @@ def get_slits(self, redo=False, write_qa=True):
 
         # Return already generated data
         if self._cached('trace', self.master_key_dict['trace']) and not redo:
-            self.tslits_dict = self.calib_dict[self.master_key_dict['trace']]['trace']
-            return self.tslits_dict
-
-        # Instantiate
-        self.edges = edgetrace.EdgeTraceSet(self.spectrograph, self.par['slitedges'],
-                                            master_key=self.master_key_dict['trace'],
-                                            master_dir=self.master_dir,
-                                            qa_path=self.qa_path if write_qa else None)
-
-        if self.reuse_masters and self.edges.exists():
-            self.edges.load()
-            self.tslits_dict = self.edges.convert_to_tslits_dict()
-        else:
-            # Build the trace image
-            self.traceImage = traceimage.TraceImage(self.spectrograph,
-                                                    files=self.trace_image_files, det=self.det,
-                                                    par=self.par['traceframe'],
-                                                    bias=self.msbias)
-            self.traceImage.build_image(bias=self.msbias, bpm=self.msbpm)
-
-            try:
-                self.edges.auto_trace(self.traceImage, bpm=self.msbpm, det=self.det,
-                                      save=self.save_masters) #, debug=True, show_stages=True)
-            except:
-                self.edges.save()
-                msgs.error('Crashed out of finding the slits. Have saved the work done to disk '
-                           'but it needs fixing.')
-                return None
-
-            # Show the result if requested
-            if self.show:
-                self.edges.show(thin=10, in_ginga=True)
-
-            # TODO: Stop-gap until we can get rid of tslits_dict
-            self.tslits_dict = self.edges.convert_to_tslits_dict()
+            self.slits = self.calib_dict[self.master_key_dict['trace']]['trace']
+            return self.slits
+
+        # Instantiate. This will load the master-frame file if it
+        # exists, and returns None otherwise.
+        self.slits = slittrace.SlitTraceSet.from_master(self.master_key_dict['trace'],
+                                                        self.master_dir, reuse=self.reuse_masters)
+        if self.slits is None:
+            # Slits don't exist or we're not resusing them
+
+            # TODO: Add a from_master function to EdgeTraceSet
+            self.edges = edgetrace.EdgeTraceSet(self.spectrograph, self.par['slitedges'],
+                                                master_key=self.master_key_dict['trace'],
+                                                master_dir=self.master_dir,
+                                                qa_path=self.qa_path if write_qa else None)
+
+            if self.reuse_masters and self.edges.exists:
+                self.edges.load()
```

Comment:
We talked this through.  The `EdgeTraceSet` object can take an image as an input instead of the `TraceImage` object.  I understand the argument for having `EdgeTraceSet` operate on a list of files, but I don't want to change this for this PR.  Let's revisit this in the future.

Reply:
Okay, can you put a TODO in edgetrace to that effect.

Reply:
Done.